cardano-testnet/test/Test/Cli/Alonzo/LeadershipSchedule.hs

```diff
@@ -499,15 +498,17 @@ hprop_leadershipSchedule = H.integration . H.runFinallies . H.workspace "alonzo"
 
   expectedLeadershipSlotNumbers <- H.noteShowM $ fmap (fmap slotNumber) $ H.leftFail $ J.parseEither (J.parseJSON @[LeadershipSlot]) scheduleJson
 
+  maxSlotExpected <- H.noteShow $ maximum expectedLeadershipSlotNumbers
+
   H.assert $ not (L.null expectedLeadershipSlotNumbers)
 
   leadershipDeadline <- H.noteShowM $ DTC.addUTCTime 90 <$> H.noteShowIO DTC.getCurrentTime
 
-  H.assertByDeadlineMCustom "Leader schedule is correct" leadershipDeadline $ do
+  -- Leader schedule is correct
+  H.byDeadlineM 10 leadershipDeadline "Wait for a leadership at least as new as the highest one we expect" $ do
```

Comment:
We have to wait until a time when all the slots we expect should exist.  The assertion doesn't do the wait.  Without the wait we would check too early and the slots won't be there.

Reply:
`"Wait for a leadership at least as new as the highest one we expect"` is opaque to me. I think here some verbosity is fine because it will make failure diagnosis easier. What about "Wait for chain to surpass all expected leadership slots". And we can add a comment as follows: "We need enough time to pass such that the expected leadership slots generated by the `leadership-schedule` command have actually occurred."

Reply:
Updated!